### ISSUE TITLE
Limited functionality in HTML5

### ISSUE THREAD

**Issue by qst0** [JujuAdams/Scribble#18]
It would seem the only way to get scribble working on an HTML5 build is to use a sprite font. Might be a problem with the way HTML5 building handles included files and sprites. I am not sure thought. Glyph usages is broken too, but [c_color], [wave] and [rainbow] work.

Errors yielded are not very useful from my perspective, even from Debug mode:

_Attempting to use sprPlayer sprite as a glyph_
```
Error: must be an array
--------------------------------------------------------------------
	function _Uf("must be an array")
	function __yy_gml_array_check_index(0, [undefined])
	function gml_Script_scribble_draw([instance], [instance], 120, 120, "[sprPlayer]")
	function gml_Object_oGame_Draw_64([instance], [instance])
	function(2112, 0, [instance], [instance])
	function(2112, 0, [instance], [instance])
	function([unknown], 2112)
	function([unknown])
	function()
	function _q13()
	function _V03()
	function _A03(4919.792)
```

**Comment by JujuAdams** [comment]
I'm yet to take this up with YoYoGames. There's a lot in Scribble that's using vertex buffers and HTML5 has historically had inconsistent support for them. It could also be the included .yy font definition files that are breaking things. It's a real mess!

**Comment by qst0** [comment]
Okay sounds about right, thanks for the quick reply! 😃

**Comment by JujuAdams** [comment]
Found numerous bugs with the HTML5 runner regarding font textures and loading data from buffers. I've filed the relevant bugs with YYG and I'm waiting back on a reponse.

**Comment by JujuAdams** [comment]
YYG have confirmed one bug - logged as "low priority" <_< - waiting on the other now.

**Comment by JujuAdams** [comment]
tbh I don't think I'll ever hear back on this one. I'm gonna leave this ticket open and I'll make sure to highlight Scribble incompatibility with HTML5 where appropriate.

**Comment by JujuAdams** [comment]
@qst0 Good news! The latest version of Scribble works in HTML5 on the latest beta runtime of GameMaker (v23.1.1.433)

**Comment by qst0** [comment]
This is awesome 🤩 
I'll have to dust off some old ideas I had with this in mind!

**Comment by DragoniteSpam** [comment]
two and a half years in the making :')
